pandas/tests/arrays/categorical/test_indexing.py

```diff
@@ -121,3 +121,11 @@ def test_get_indexer_non_unique(self, idx_values, key_values, key_class):
 
             tm.assert_numpy_array_equal(expected, result)
             tm.assert_numpy_array_equal(exp_miss, res_miss)
+
+
+class TestSetIndex(object):
```

Comment:
put in pandas/tests/frame/test_alter_axes.py.

put in a full comparison on the result table and use assert_fram_equal

Reply:
`df = DataFrame([[2, 1, 2], [4, (1, 2), 3]]).set_index([0, 1])` gives me
Out[90]: 
             2
0 1        
2 1        2
4 (1, 2)  3

How I construct this directly from the DataFrame constructor? The first fow seems empty?
df.to_dict() gives `{2: {(2, 1): 2, (4, (1, 2)): 3}}`

I tried:
DataFrame(['',2,3],
          columns=['2'],
          index=[['0','2','4'], ['1','1','(1,2)']])

but it's not equal (even though it looks equal), but 
DataFrame shape mismatch
[left]:  (2, 1)
[right]: (3, 1)

Reply:
> but it's not equal (even though it looks equal), but

Yes, they look equal, but you can see from how you built them they are not. ``[0, 1]`` are the index names, not index values.

```
pd.DataFrame([2,3], columns=['2'],
             index=pd.MultiIndex.from_arrays([['2','4'], ['1','(1,2)']], names=[0, 1]))
```
is what you're looking for.

Reply:
> I also tried

The original ``df`` has no strings at all, just integers. Why are you putting strings in the other (e.g. ``'2'``, ``'(1,2)'``?!)

Reply:
Removing the two strings I get `ValueError: setting an array element with a sequence.`

Reply:
> Removing the two strings I get ValueError: setting an array element with a sequence.

OK, ``pd.MultiIndex.from_tuples([(2, 1), (4, (1,2))], names=[0, 1])``